lib/srv/desktop/tdp/proto.go

```diff
@@ -915,12 +915,19 @@ func decodeSharedDirectoryCreateRequest(in peekReader) (SharedDirectoryCreateReq
 type SharedDirectoryCreateResponse struct {
 	CompletionID uint32
 	ErrCode      uint32
+	Fso          FileSystemObject
 }
 
 func (s SharedDirectoryCreateResponse) Encode() ([]byte, error) {
 	buf := new(bytes.Buffer)
 	buf.WriteByte(byte(TypeSharedDirectoryCreateResponse))
-	binary.Write(buf, binary.BigEndian, s)
+	binary.Write(buf, binary.BigEndian, s.CompletionID)
+	binary.Write(buf, binary.BigEndian, s.ErrCode)
+	fsoEnc, err := s.Fso.Encode()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	binary.Write(buf, binary.BigEndian, fsoEnc)
```

Comment:
`fsoEnc` is just bytes at this point, right? So `buf.Write(fsoEnc)` would make more sense.